File: LeetCode/495.cpp
#include <iostream>
#include <vector>
#include <cmath>
using namespace std;

class Solution {
public:
    int findPoisonedDuration(vector<int>& timeSeries, int duration) {
        int ans = 0, stop = 0;
        for (vector<int>::size_type i = 0; i < timeSeries.size(); ++i){
            ans += duration - max(stop - timeSeries[i], 0);
            stop = timeSeries[i] + duration;
        }
        return ans;
    }
};

File: LeetCode/495.py
class Solution(object):
    def findPoisonedDuration(self, timeSeries, duration):
        """
        :type timeSeries: List[int]
        :type duration: int
        :rtype: int
        """
        ans, stop = 0, 0
        for t in timeSeries:
            ans += duration - max(stop - t, 0)
            stop = t + duration
        return ans